webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubrics-shared-list.js

```diff
@@ -54,7 +54,25 @@ export class SakaiRubricsSharedList extends RubricsElement {
     var params = {"projection": "inlineRubric"};
 
     SakaiRubricsHelpers.get("/rubrics-service/rest/rubrics/search/shared-only", token, { params })
-      .then(data => this.rubrics = data._embedded.rubrics );
+    .then( (data) => {
+       this.rubrics = data._embedded.rubrics 
+
+       // To sort the rubrics correctly we need the user and the site names in the arrays, not the ids
+       this.rubrics = this.rubrics.map( (rubric) => {
+         let metadata = rubric.metadata;
+         const creatorId = metadata.creatorId;
+         const siteId = metadata.ownerId;
+         fetch('/sakai-ws/rest/sakai/getUserDisplayName?sessionid=' + sakaiSessionId + '&eid=' + creatorId)
+         .then( (response) => response.text())
+         .then( (name) => metadata.creatorName = name);
+         fetch('/sakai-ws/rest/sakai/getSiteTitle?sessionid=' + sakaiSessionId + '&siteid=' + siteId)
+         .then( (response) => response.text())
+         .then( (name) => metadata.siteName = name);
+         rubric.metadata = metadata;
+         return rubric;
```

Comment:
all this code needs pulling into a utils class. I think rubrics has one already.

Reply:
awww yeah, that makes sense, included a separate commit for that.